espnet2/enh/diffusion_enh.py

```diff
@@ -0,0 +1,156 @@
+"""Enhancement model module."""
+import contextlib
+from typing import Dict, List, Optional, OrderedDict, Tuple
+
+import torch
+from typeguard import check_argument_types
+
+from espnet2.enh.decoder.abs_decoder import AbsDecoder
+from espnet2.enh.diffusion.abs_diffusion import AbsDiffusion
+from espnet2.enh.encoder.abs_encoder import AbsEncoder
+from espnet2.enh.espnet_model import ESPnetEnhancementModel
+from espnet2.enh.extractor.abs_extractor import AbsExtractor
+from espnet2.enh.loss.criterions.tf_domain import FrequencyDomainLoss
+from espnet2.enh.loss.criterions.time_domain import TimeDomainLoss
+from espnet2.enh.loss.wrappers.abs_wrapper import AbsLossWrapper
+from espnet2.torch_utils.device_funcs import force_gatherable
+from espnet2.train.abs_espnet_model import AbsESPnetModel
+
+EPS = torch.finfo(torch.get_default_dtype()).eps
+
+
+class ESPnetDiffusionModel(ESPnetEnhancementModel):
+    """Target Speaker Extraction Frontend model"""
+
+    def __init__(
+        self,
+        encoder: AbsEncoder,
+        diffusion: AbsDiffusion,
+        decoder: AbsDecoder,
+        # loss_wrappers: List[AbsLossWrapper],
```

Comment:
Do you disable multi-task losses for diffusion-based methods?

Reply:
Yes, it's disabled.